apps/src/code-studio/assets/SpriteUpload.jsx

```diff
@@ -88,9 +127,30 @@ export default class SpriteUpload extends React.Component {
     this.setState({spriteAvailability: event.target.value, category: ''});
   };
 
+  handleAliasChange = event => {
+    let aliases = event.target.value;
+    aliases = aliases.split(',');
+    let processedAliases = aliases.map(alias => alias.trim());
+    this.setState({aliases: processedAliases});
+  };
+
+  generateMetadata = () => {
+    const {filename, aliases} = this.state;
+    let image = document.getElementById('sprite-image-preview');
+    let metadata = {};
+    metadata['name'] = filename;
+    metadata['aliases'] = aliases;
+    metadata['frameCount'] = 1;
+    metadata['frameSize'] = {x: image.width, y: image.height};
+    metadata['looping'] = true;
+    metadata['frameDelay'] = 2;
```

Comment:
nit: i think it would be clearer to create the object directly rather than setting each property separately, e.g.:

```javascript
const metadata = {
  name: filename,
  ...
}
```